http/fetch/archive_fetcher.go

```diff
@@ -72,19 +87,41 @@ func NewArchiveFetcherWithLogger(retries, maxDownloadSize, maxUntarSize int, hos
 		httpClient.Logger = logger
 	}
 
-	return &ArchiveFetcher{
+	a := &ArchiveFetcher{
 		httpClient:        httpClient,
 		maxDownloadSize:   maxDownloadSize,
 		maxUntarSize:      maxUntarSize,
 		hostnameOverwrite: hostnameOverwrite,
 	}
+
+	for _, opt := range opts {
+		opt(&a.opts)
+	}
+
+	return a
 }
 
-// Fetch downloads, verifies and extracts the tarball content to the specified directory.
+// Fetch downloads and verifies the tarball digest. If WithSkipUntar() was used, the tarball file is
+// stored unextracted in the given path, otherwise the tarball content is extracted into the given path.
 // If the file server responds with 5xx errors, the download operation is retried.
 // If the file server responds with 404, the returned error is of type ErrFileNotFound.
 // If the file server is unavailable for more than 3 minutes, the returned error contains the original status code.
-func (r *ArchiveFetcher) Fetch(archiveURL, digest, dir string) error {
+func (r *ArchiveFetcher) Fetch(archiveURL, digest, path string) error {
+	var f *os.File
+	var err error
+	if !r.opts.skipUntar {
```

Comment:
maybe it better to use a temporary file for all operations and if we want to skip uncompressing just use `err = os.Rename(f.Name(), path)`

Reply:
In my use case they won't be in the same directory

Reply:
The target file will be in an `emptyDir` volume, and not under `/tmp` like in `os.CreateTemp()`